### ISSUE TITLE
Wrong control flow of break/continue expanded from a macro

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#7016]
## Environment

* **IntelliJ Rust plugin version:** 0.3.144.3748-211-nightly
* **Rust toolchain version:** 1.51.0-nightly (04caa632d 2021-01-30) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2021.1 Beta Ultimate Edition (IU-211.6556.6)
* **Operating system:** macOS 11.1
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Control-flow analysis should consider `break`/`continue` expressions expanded from a macro. Currently, macro-expanded `break`/`continue` expressions are just ignored in terms of control flow. This affects `Liveness analysis` and `Borrow checker` and leads to false-positive annotations.

## Steps to reproduce
```Rust
macro_rules! break_macro {
    () => { break };
}
fn main() {
    let x = 1; // false positive "Variable x is never used"
    loop {
        break_macro!();
    }
    x;
}
```
